### DISCUSSION TITLE
Consistent "latest version" download links

### DISCUSSION THREAD

**Discussion by jontjs** [adminerevo/adminerevo#98] [Ideas]
Following on from the completed discussion at https://github.com/adminerevo/adminerevo/discussions/84...

It would be very useful for automation (e.g. installing adminerevo when spinning up a development environment) if the downloads didn't have the version number in them. 

Currently when spinning up a development environment, our scripts can simply fetch the latest version of the "historical" adminer from https://www.adminer.org/latest-mysql-en.php but the closest we can get with adminerevo is to fetch https://download.adminerevo.org/latest/adminer/adminer-4.8.3-mysql-en.zip which requires knowledge of the current version at the time. 

It would also be slightly-helpful if the downloads were not zipped, but that's an easy thing for us to work around so I can respect your need to save disk space and/or bandwidth.

**Comment by LionelLaffineur** [comment]
Dear @jontjs, your request has sense (for the release number).
@Roy-Orbison, would you like to comment about this ?
About the fact it is a zip, it's because my hosting doesn't allow me to let download PHP files but always executes them.
Regards,
L.

**Comment by Letme** [comment]
It does not make sense to drop released version - it is a bad practice to automatically follow the latest version especially for backwards incompatible changes.
You can always query github API to get the latest release if that is really what you want.

**Reply by jontjs** [reply]
I completely agree that https://download.adminerevo.org/ needs to keep the versioned directories for people who need those so that they can get a predictable version.

I just think that, presumably, the reason for having the "latest" directory is that you can use it to easily get whatever is the latest version at that moment. But the filenames currently within it prevent use of it in that way, because they will change whenever a new "latest" is released.

As for backwards compatibility - that's not a concern in our use-case, but I don;t think my suggestions would harm anybody who needs need to be certain of the version they get.

**Reply by Letme** [reply]
You can always create symlink for example adminer.php -> adminer-1-2-3.php if you want to serve it that way. Then you simply move the symlink to a new version and delete the older ones (to avoid leaving open backdoors old versions might have).

**Comment by devinemke** [comment]
Here is how I have solved this problem (for my needs anyway). As @Letme suggested, fetch the latest AdminerEvo version programmatically via a simple shell script (in which the `$version` variable is the current number, whatever it may be):
```
version=$(curl https://api.github.com/repos/adminerevo/adminerevo/releases/latest | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
version=${version:1}
curl -vo adminer-$version-mysql-en.zip https://download.adminerevo.org/latest/adminer/adminer-$version-mysql-en.zip
unzip -v adminer-$version-mysql-en.zip
```
Hope this helps.

**Reply by Letme** [reply]
Now you are just missing `ln -s adminer.php adminer-$version-mysql-en.php` (arguments might be swapped) and you have steady access to adminer via adminer.php (or link to index.php and then use subdomain). This helps keeping steady interface while in background maintaining explicit versioned reference.

**Comment by LionelLaffineur** [comment]
I have to think about ... I'm limited by my hosting provider.  The version number is generated in the filename by the compile.php script.  It could be that I remove it when zipping the file, I don't know.

**Reply by jontjs** [reply]
Hi Lionel,

It sounds more awkward to change the filenames than I guessed - I shall use a combination of the suggestions from @devinemke and @Letme .

many thanks for pickimng up development on this very useful tool!

regards,
Jon

**Comment by LionelLaffineur** [comment]
Reading all your comments, I have changed my generation scripts and re-generated the files.
Since the version number is in the directory, it is useless to have it in the filename.  I was using the file with the name generated by compile.php, now I do a rename in the script.
There is now the possibility to use always the same URL to download the latest version with a certain driver/language.

Regards,
L.

**Reply by jontjs** [reply]
Thank you very much Lionel, it is much appreciated!

best wishes,
Jon
